addons/base_import_module/models/ir_module.py

```diff
@@ -117,8 +128,36 @@ def import_zipfile(self, module_file, force=False):
                             _logger.exception('Error while importing module')
                             errors[mod_name] = exception_to_unicode(e)
                 finally:
-                    addons.module.ad_paths.remove(module_dir)
+                    module.ad_paths.remove(module_dir)
         r = ["Successfully imported module '%s'" % mod for mod in success]
         for mod, error in errors.items():
             r.append("Error while importing module '%s': %r" % (mod, error))
         return '\n'.join(r), module_names
+
+
+def _is_studio_custom(path):
+    """
+    Checks the to-be-imported records to see if there are any references to
+    studio, which would mean that the module was created using studio
+
+    Returns True if any of the records contains a context with the key
+    studio in it, False if none of the records do
+    """
+    path = os.path.join(path, 'data')
+    views = [z for x, y, z in os.walk(path)][0]
+    views = [f for f in views if os.path.splitext(f)[1].lower() == '.xml']
+
+    for view in views:
+        with open(os.path.join(path, view), 'rb') as f:
+            raw = f.read()
+            root = lxml.etree.fromstring(raw)
+
+        for record in root:
+            # ast.literal_eval is like eval(), but safer
+            # context is a string representing a python dict
+            ctx = ast.literal_eval(record.get('context'))
```

Comment:
This may crash in case `record` has no attribute `context`.
Add a `try`/`except` guard around the loop body.